### PR TITLE
Alex/4952 hotstuff error on equivocating proposal

### PR DESCRIPTION
**Pull request by AlexHentschel**
fixes an uncovered edge case in the VoteAggregator:
* a double proposal can crashes consensus nodes, which are processing the proposal 

We saw this happening to one of our partners (see https://github.com/dapperlabs/flow-go/issues/4952), who wiped the state of their collector node. Purging the collectors persisted state led to the collector repeating old views and proposing (different) block for views where it had already proposed before. Luckily, this only crashed the collector itself; but only because the other replicas were already so far ahead that they would just drop the equivocating node's proposals. If other committee members would have processed the equivocating blocks, they would have crashed as well.

This PR:
* fixes the issue 
* adds a test case to the VoteAggregator to test correct handling of this particular edge case
* also includes a logging extension when receiving an Execution Receipt: the logs now include the height of the block the Exec Receipt pertains to 

#### Implementation Details
* The problem in the implementation _prior_ to this fix was the placement of the double-voting check within the `VoteAggregator.validateAndStoreIncorporatedVote` method https://github.com/onflow/flow-go/blob/6fae059a61f4ff7d2e125d24a22007eb625ead7d/consensus/hotstuff/voteaggregator/vote_aggregator.go#L275-L280 Essentially, we are treating double voters as invalid, which caused the following sanity check to fail: https://github.com/onflow/flow-go/blob/6fae059a61f4ff7d2e125d24a22007eb625ead7d/consensus/hotstuff/voteaggregator/vote_aggregator.go#L196-L199 
* In the updated version, we consider double votes still as valid votes. This removes the edge case. The potential downside is that we would be counting votes from equivocating nodes for different blocks towards the required minimum stake (instead of dropping them). However, also previously, we would only drop equivocating votes on a best-effort basis. Specifically, we did _not_ go back and remove equivocating votes if we already added them to the VotingStatus of a block; we just disregarded subsequent equivocating votes. 
* Including equivocating votes in building the QC for the respective block does _not_ reduce security. The main deterrent against equivocation is the slashing, which only requires detection of double voting.

### CODE REVIEW

**Review by zhangchiqing — APPROVED** [review]
Good catch 👍

File: consensus/hotstuff/voteaggregator/vote_aggregator_test.go [hunk 1]
```diff
@@ -575,6 +575,45 @@ func (as *AggregatorSuite) TestDuplicateVotesBeforeBlock() {
...
+	bp2 := newMockBlock(as, testView, as.participants[0].NodeID)
+
+	// Each of the blocks contain the proposer's signature. Hence, the proposer
+	// votes for both of its conflicting blocks => expect double vote notification
+	as.notifier.On("OnDoubleVotingDetected", bp1.ProposerVote(), bp2.ProposerVote()).Return()
+
+	as.aggregator.StoreProposerVote(bp1.ProposerVote())
+	qc, built, err := as.aggregator.BuildQCOnReceivedBlock(bp1.Block)
+	require.Nil(as.T(), qc)
+	require.False(as.T(), built)
```
Comment (jordanschalm) [inline consensus/hotstuff/voteaggregator/vote_aggregator_test.go:597]: Is this false just because there is only one vote for the block? Or is it related to the fix as well?

**Review by jordanschalm — APPROVED** [review]

File: consensus/hotstuff/voteaggregator/vote_aggregator_test.go (hunk 1 above)
Comment (AlexHentschel) [reply consensus/hotstuff/voteaggregator/vote_aggregator_test.go:597]: In the test setup, we have 7 consensus participants, i.e. we require 5 votes for a supermajority. The fix didn't change this logic.
